rust/template/differential_datalog/src/program/mod.rs

```diff
@@ -1026,22 +1023,26 @@ impl Program {
         // Clone the program so that it can be moved into the timely computation
         let program = Arc::new(self.clone());
         let timely_config = config.timely_config()?;
-        let (worker_config, profiling_data) = (config, profiling_rig.profiling_data.clone());
+        let (worker_config, profiling_data) =
+            (config.clone(), profiling_rig.profiling_data.clone());
 
         // Start up timely computation.
         let worker_guards = timely::execute(
             timely_config,
             move |worker: &mut Worker<Allocator>| -> Result<_, String> {
                 let worker = DDlogWorker::new(
                     worker,
-                    worker_config,
+                    worker_config.clone(),
                     program.clone(),
                     profiling_data.clone(),
                     Arc::clone(&request_recv),
                     Arc::clone(&reply_send),
                 );
 
-                worker.run()
+                worker.run().map_err(|e| {
+                    eprintln!("Worker thread failed: {}", e);
+                    e
+                })
```

Comment:
This will make things fail if `TIMELY_WORKER_ADDR` happens to be set in the environment since timely sets some stuff up by default, I ran into the same issue in ddshow and [this was my workaround](https://github.com/Kixiron/ddshow/blob/master/src/main.rs#L70-L82). As a plus doing this also makes the stuff [here](https://github.com/vmware/differential-datalog/blob/43f7d2c34dee916ff5a9ccb0bcdd540723f82e3b/rust/template/differential_datalog/src/program/worker.rs#L524-L527) redundant

Reply:
Ah, glad you caught this.  So the solution is to use `execute_from` instead of `execute`?

Reply:
Yep, it's a little annoying since you have to do stuff manually but that's the gist